build_tools/travis/test_pytest_soft_dependency.sh

```diff
@@ -0,0 +1,18 @@
+##!/bin/bash
+
+set -e
+
+if [[ "$CHECK_PYTEST_SOFT_DEPENDENCY" == "true" ]]; then
```

Comment:
Out of curiosity why did you move this out of `test_script.sh`?

Reply:
The original `test_script.sh` order was:

1. Run pytest scikit-learn modules.
2. Run doc-test.
3. Run check_pytest_soft_dependency, which removes pytest.

If `check_pytest_soft_dependency` was kept in `test_script.sh`, pytest would be removed before `doc-test` gets to run.